src/main.c

```diff
@@ -391,7 +391,7 @@ poe_cmd_power_mgmt_mode(unsigned char mode)
 
 /* 0x18 - Set global power budget */
 static int
-poe_cmd_global_power_budget(int budget, int guard)
+poe_cmd_global_power_budget(float budget, float guard)
```

Comment:
`write16_be` [expects](https://github.com/Hurricos/realtek-poe/blob/db485c85b2c6c997049b81c44a3c0d6d57d1728c/src/main.c#L400) a `uint16_t`, yet this change causes it to now be passed a float, twice ([`float * int = float`](https://stackoverflow.com/a/17057689)).

I'd ask you to cast, but I'm more alarmed by the fact that this code compiles and works.

Reply:
There's an implicit int conversion. That is valid C.

Now there is an issue if the result is outside of [0, UINT16_MAX]. But that issue also existed before this patch,

Reply:
Here's the wording from the C standard, for those playing along at home:

```
6.3.1.4 Real floating and integer
1 When a finite value of real floating type is converted to an integer type other than _Bool,
the fractional part is discarded (i.e., the v alue is truncated to ward zero). If the v alue of
the integral part cannot be represented by the integer type, the behavior is undefined.
```

Reply:
*That's* what I'm missing :^)